### ISSUE TITLE
Implementation of stochastic and batch grandient descent in python

### ISSUE THREAD

**Issue by jfofigrgk** [jfofigrgk/Akshith-P-G#2]
import pandas as pa
import numpy as nu 
import matplotlib.pyplot as plt 
from sklearn.preprocessing import MinMaxScaler
import random
df = pa.read_csv("homeprices_banglore.csv")
df.sample(5)
z = MinMaxScaler()
red_x = z.fit_transform(df.drop("price",axis="columns"))
red_y = z.fit_transform(df["price"].values.reshape(df.shape[0],1))
red_y = red_y.reshape(20,)
random_indices = nu.random.permutation(5)
red_x=red_x[random_indices]
for  j in range(0,20,5):
     xj = red_x[j:j+5]
     def mini_batch_gradient_descent(X, y_true, epochs = 100, batch_size = 5, learning_rate = 0.01):
    number_of_features = X.shape[1]
    # numpy array with 1 row and columns equal to number of features. In 
    # our case number_of_features = 3 (area, bedroom and age)
    w = nu.ones(shape=(number_of_features)) 
    b = 0
    total_samples = X.shape[0] # number of rows in X
    
    if batch_size > total_samples: # In this case mini batch becomes same as batch gradient descent
        batch_size = total_samples
        
    cost_list = []
    epoch_list = []
    for i in range(epochs):
        random_indices = nu.random.permutation(batch_size)
        x_temp = X[random_indices]
        y_temp = y_true[random_indices]
        for  j in range(0,total_samples,batch_size):
            Xj = x_temp[j:j+batch_size]
            yj = y_temp[j:j+batch_size]
            y_predicted = nu.dot(w, Xj.T) + b
            
            w_grad = -(2/len(Xj))*(Xj.T.dot(yj-y_predicted))
            b_grad = -(2/len(Xj))*nu.sum(yj-y_predicted)
            
            w = w - learning_rate * w_grad
            b = b - learning_rate * b_grad
                
            cost = nu.mean(nu.square(yj-y_predicted)) # MSE (Mean Squared Error)
        
        if i%10==0:
            cost_list.append(cost)
            epoch_list.append(i)
        return w, b, cost, cost_list, epoch_list
